buzzmobile/tests/test_utils/launch_tools.py

```diff
@@ -44,13 +94,17 @@ def launcher(func):
         def new_test(self):
             """Wrapper around the user provided test that runs a launch file.
             """
-            launch = ROSLauncher(full_name)
+            os.environ['ROS_MASTER_URI'] = self.rosmaster_uri
+            launch = ROSLauncher(full_name, port=self.port)
             launch.start()
-            global _LAUNCHER # pylint: disable=global-statement
-            _LAUNCHER = launch
+            if self.port in _LAUNCHER:
+                raise Exception('You are using these incorrectly')
```

Comment:
'Rosmaster port {} already in use. Make sure with_file_name is the first decorator'

or something like that